src/Connection_Oracle.php

```diff
@@ -32,27 +30,19 @@ public function __construct($properties = [])
     /**
      * Return last inserted ID value.
      *
-     * Few Connection drivers need to receive Model to get ID because PDO doesn't support this method.
+     * Few Connection drivers need to receive sequence name to get ID because PDO doesn't support this method.
      *
-     * @param Model Optional data model from which to return last ID
-     * @param string Optional sequence name from which to return last ID (this takes precedence)
+     * @param string $sequence Optional sequence name from which to return last ID
      *
      * @return mixed
      */
-    public function lastInsertID(Model $m = null, string $sequence = null)
+    public function lastInsertID(string $sequence = null)
     {
-        $seq = $sequence ?? ($m && $m->sequence ? $m->sequence : null);
-
-        if ($seq) {
-            return $this->dsql()->mode('seq_currval')->sequence($seq)->getOne();
-        } elseif ($m instanceof Model) {
-            // otherwise we have to select max(id_field) - this can be bad for performance !!!
-            // Imants: Disabled for now because otherwise this will work even if database use triggers or
-            // any other mechanism to automatically increment ID and we can't tell this line to not execute.
-            //return $this->expr('SELECT max([field]) FROM [table]', ['field'=>$m->id_field, 'table'=>$m->table])->getOne();
+        if ($sequence) {
+            return $this->dsql()->mode('seq_currval')->sequence($sequence)->getOne();
         }
 
         // fallback
-        return parent::lastInsertID($m, $sequence);
+        return parent::lastInsertID($sequence);
```

Comment:
No need for fallback, citing from PHP doc:
> PDO_PGSQL requires you to specify the name of a sequence object

Reply:
This ir Oracle not PgSQL. So if we have sequence set, then we call SELECT $sequence.CURRVAL FROM dual otherwise fallback to default lastInsertID. For example, id is set by some DB trigger.

Reply:
I do not know Oracle behaviour on this... If there is usecase, then ok, if not, it should be removed.

Reply:
There can be. I have used Oracle some years ago (not with agile toolkit).